packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -8,35 +8,291 @@ import { Any } from '@agoric/cosmic-proto/google/protobuf/any.js';
+  const denomAmount = {
+    denom,
+    value: amount.value,
+  };
+
+  await kit.keeper.getAccount(YieldProtocol.Aave).transfer(
+    {
+      value: gmpAddresses.AXELAR_GMP,
+      encoding: 'bech32',
+      chainId,
+    },
+    denomAmount,
+    { memo: JSON.stringify(memo) },
+  );
+
+  seat.exit();
+};
+
+const supplyToAave = async (
+  orch: Orchestrator,
+  seat: ZCFSeat,
+  ctx: {
+    axelarChainsMap: AxelarChainsMap;
+    chainHubTools: {
+      getDenom: (brand: Brand) => Denom | undefined;
+    };
+    contractAddresses: EVMContractAddresses;
+    kit: PortfolioKit;
+  },
+  gmpArgs: {
+    destinationEVMChain: AxelarChain;
+    gasAmount: Amount<'nat'>;
+    transferAmount: BigInt;
+  },
+) => {
+  const { destinationEVMChain, transferAmount, gasAmount } = gmpArgs;
+  const { kit, contractAddresses } = ctx;
+
+  const remoteEVMAddress = await kit.keeper.getRemoteAddress();
+
+  await sendGmp(
+    orch,
+    seat,
+    ctx,
+    harden({
+      destinationAddress: contractAddresses.factory,
+      destinationEVMChain,
+      type: AxelarGMPMessageType.ContractCall,
+      amount: gasAmount,
+      contractInvocationData: [
+        {
+          functionSignature: 'approve(address,uint256)',
+          args: [contractAddresses.aavePool, transferAmount],
+          target: contractAddresses.usdc,
+        },
+        {
+          functionSignature: 'supply(address,uint256,address,uint16)',
+          args: [contractAddresses.usdc, transferAmount, remoteEVMAddress, 0],
+          target: contractAddresses.aavePool,
+        },
+      ],
+    }),
+  );
+};
+
+const withdrawFromAave = async (
+  orch: Orchestrator,
+  seat: ZCFSeat,
+  ctx: {
+    axelarChainsMap: AxelarChainsMap;
+    chainHubTools: {
+      getDenom: (brand: Brand) => Denom | undefined;
+    };
+    contractAddresses: EVMContractAddresses;
+    kit: PortfolioKit;
+  },
+  gmpArgs: {
+    destinationEVMChain: AxelarChain;
+    gasAmount: Amount<'nat'>;
+    withdrawAmount: BigInt;
+  },
+) => {
+  const { destinationEVMChain, withdrawAmount, gasAmount } = gmpArgs;
+  const { kit, contractAddresses } = ctx;
+
+  const remoteEVMAddress = await kit.keeper.getRemoteAddress();
+
+  await sendGmp(
+    orch,
+    seat,
+    ctx,
+    harden({
+      destinationAddress: contractAddresses.factory,
+      destinationEVMChain,
+      type: AxelarGMPMessageType.ContractCall,
+      amount: gasAmount,
+      contractInvocationData: [
+        {
+          functionSignature: 'withdraw(address,uint256,address)',
+          args: [contractAddresses.usdc, withdrawAmount, remoteEVMAddress],
+          target: contractAddresses.aavePool,
+        },
+      ],
+    }),
+  );
+};
+
+const supplyToCompound = async (
+  orch: Orchestrator,
+  seat: ZCFSeat,
+  ctx: {
+    axelarChainsMap: AxelarChainsMap;
+    chainHubTools: {
+      getDenom: (brand: Brand) => Denom | undefined;
+    };
+    contractAddresses: EVMContractAddresses;
+    kit: PortfolioKit;
+  },
```

Comment:
Please use `Pick<PortfolioContext, 'axelarChainsMap' | ...>`

Reply:
and don't try to pass `kit` inside `ctx`. In fact, pass `keeper` as its own arg.